en/codegen/layout-codegen.md

```diff
@@ -25,43 +25,112 @@ In the below image the developer would select the Sketch Groups `peopleList` and
-Add the required Dialog, Toast, and Snackbar to the drawing, configure them as you would any other element, then hide them as below. This form would be the one the developer would select to code generate.
+Add the required Dialog, Toast, and Tooltip to the drawing, configure them as you would any other element, then hide them as below. This form would be the one the developer would select to code generate.
 The hiding of element in the object panel has no effect on code generation, code generation will generate all elements here, hidden or not.
-Dialog, Toast, and Snackbar elements will be placed at the bottom of the components HTML and not mixed in with other elements as they have dynamic runtime placement when brought into view.
+Dialog, Toast, and Tooltip elements will be placed at the bottom of the components HTML and not mixed in with other elements as they have dynamic runtime placement when brought into view.
 
 <img class="responsive-img" src="../images/layout_codegen17.png" />
 
 The below image is an example of a designer state drawing, this drawing should not be used for code generation as it has an extra color element to simulate the gray overlay that the Dialog would place at runtime.
 
 <img class="responsive-img" src="../images/layout_codegen18.png" />
 
+### Positioning Layouts
+Layout position is relative to their parent. During layout building if a group has one child, then the group takes the dimensions of the child (position and size). The child assumes position (0,0) relative to its parent. When a new child needs to be included to a group it may affect the dimensions of the group, if this is the case then all other children in the group are going to be updated.
+
+If a group remains with one child, the structure is optimized and the group is omitted.
+
+### Group size
+  - Columns - the width of a column represents the width of its children. For the height the column fully fills the vertical space determined by its parent.
+  - Rows - the height of a row represents the height of its children. For the width the row fully fills the horizontal space determined by its parent.
+
+> [!Note]
+> Additional paddings/margins are applied after justification is resolved based on the justification type.
+
+### Justification and alignment
+Rows receive `row` value for `flex-direction` css property and columns receive `column`.
+
+Code generation will try to match the most appropriate value for `justify-content` property in css. Using the position and size of the elements in a group, the whole group can justify its elements in the beginning, at the end, center them ant etc.
+Depending on the spaces between children and from the edges of the parent. Also on the counter axis will try to determine the most suitable value for `align-items` css property.
```

Comment:
This paragraph looks weird when previewed.

"Depending on the spaces between children and from the edges of the parent" should be part of another sentence